### PR TITLE
add available weights to SMP

### LINKED ISSUE #517: Add support for weights paths to the SEMANTIC_SEGMENTATION_BACKBONES registry

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

Fixes #517 #518 #519

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — COMMENTED** [review]
Awesome, looking really neat :smiley: I think the only thing that's missing is checking if pretrained is a string when loading the head and using that for the weights path

**Comment by aniketmaurya** [comment]
> Awesome, looking really neat 😃 I think the only thing that's missing is checking if pretrained is a string when loading the head and using that for the weights path

updated ✅

File: flash/image/segmentation/heads.py [hunk 1]
```diff
@@ -33,18 +35,20 @@
...
 
         if head not in SMP_MODELS:
             raise NotImplementedError(f"{head} is not implemented! Supported heads -> {SMP_MODELS.keys()}")
 
         encoder_weights = None
-        if pretrained:
+        if pretrained is True:
             encoder_weights = "imagenet"
+        elif isinstance(pretrained, str):
+            encoder_weights = pretrained
```
Comment (ethanwharris) [inline flash/image/segmentation/heads.py:51]: Maybe better to switch:
```suggestion
        if isinstance(pretrained, str):
            encoder_weights = pretrained
        elif pretrained:
            encoder_weights = "imagenet"
```

**Review by ethanwharris — APPROVED** [review]
Awesome!

**Pull request merged**